File: app/dbmodels/AssetVirtualMachine.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'AssetVirtualMachine',
    'menu'          => ''
);
$table['fields'] = array(
    'id'                        => DBModels::type('increments', 
                                                  array('visible' => false)),
    'asset_id'                  => DBModels::type('integer', 
                                                  array('visible' => false)),
    'name'                      => DBModels::type('string'),
    'comment'                   => DBModels::type('text'),
    'uuid'                      => DBModels::type('string'),
    'virtual_machine_state_id'  => DBModels::type('integer'),
    'virtual_machine_system_id' => DBModels::type('integer'),
    'virtual_machine_type_id'   => DBModels::type('integer'),
    'vcpu'        => DBModels::type('integer'),
    'is_deleted'       => DBModels::type('boolean'),
    'is_dynamic'       => DBModels::type('boolean'),
    'ram'         => DBModels::type('string'),

);

$table['relationships'] = array(
    'asset'     => array(
        'type'  => 'belongsTo',
        'item'  => 'Asset'),
    'virtual_machine_state'  => array(
        'type'  => 'belongsTo',
        'item'  => 'VirtualMachineState'),
    'virtual_machine_system'  => array(
        'type'  => 'belongsTo',
        'item'  => 'VirtualMachineSystem'),
    'virtual_machine_type'  => array(
        'type'  => 'belongsTo',
        'item'  => 'VirtualMachineType'),
);


include "_commonentitylink.php";
include "_commonrecursiveentitylink.php";

File: fusionglpi/database/asset_virtualmachine.py
from fusionglpi import db


class AssetVirtualmachine(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    asset_id = db.Column(db.Integer, db.ForeignKey('asset.id'))
    name = db.Column(db.String(255, None, True))
    comment = db.Column(db.Text)
    uuid = db.Column(db.String(255, None, True))
    virtualmachine_state_id = db.Column(
        db.Integer,
        db.ForeignKey('virtualmachine_state.id')
    )
    virtualmachine_system_id = db.Column(
        db.Integer,
        db.ForeignKey('virtualmachine_system.id')
    )
    virtualmachine_type_id = db.Column(
        db.Integer,
        db.ForeignKey('virtualmachine_type.id')
    )
    vcpu = db.Column(db.Integer, nullable=False, server_default="0")
    is_deleted = db.Column(db.Boolean, nullable=False, server_default="0")
    is_dynamic = db.Column(db.Boolean, nullable=False, server_default="0")
    ram = db.Column(db.String(255, None, True))
